src/systems/systems/src/air_conditioning/acs_controller.rs

```diff
@@ -0,0 +1,1808 @@
+    flow_selector_position: OvhdFlowSelector,
+    pack_in_start_condition: bool,
+    single_pack_operation: bool,
+}
+
+impl PackFlowController {
+    const PACK_START_TIME: f64 = 30.;
+    const PACK_START_FLOW_LIMIT: f64 = 100.;
+    const APU_SUPPLY_FLOW_LIMIT: f64 = 120.;
+    const ONE_PACK_FLOW_LIMIT: f64 = 120.;
+    const FLOW_REDUCTION_LIMIT: f64 = 80.;
+    const BACKFLOW_LIMIT: f64 = 80.;
+
+    const FLOW_CONSTANT_C: f64 = 0.5675; // kg/s
+    const FLOW_CONSTANT_XCAB: f64 = 0.00001828; // kg(feet*s)
+
+    fn new(context: &mut InitContext) -> Self {
+        Self {
+            apu_bleed_valve_open_id: context.get_identifier("APU_BLEED_AIR_VALVE_OPEN".to_owned()),
+            apu_rpm_id: context.get_identifier("APU_N_RAW".to_owned()),
+            crossbleed_id: context.get_identifier("KNOB_OVHD_AIRCOND_XBLEED_Position".to_owned()),
+            ditching_id: context.get_identifier("OVHD_PRESS_DITCHING_PB_IS_ON".to_owned()),
+            engine_1_state_id: context.get_identifier("ENGINE_STATE:1".to_owned()),
+            engine_2_state_id: context.get_identifier("ENGINE_STATE:2".to_owned()),
+            eng_1_fire_id: context.get_identifier("Fire_ENG1_Agent1_Discharge".to_owned()),
+            eng_2_fire_id: context.get_identifier("Fire_ENG2_Agent1_Discharge".to_owned()),
+            ovhd_flow_selector_id: context
+                .get_identifier("KNOB_OVHD_AIRCOND_PACKFLOW_Position".to_owned()),
+
+            pack_flow_id: context.get_identifier("COND_PACK_FLOW".to_owned()),
+
+            flow_demand: Ratio::new::<percent>(0.),
+            absolute_flow: MassRate::new::<kilogram_per_second>(0.),
+            fcv_1_open_allowed: false,
+            fcv_2_open_allowed: false,
+            should_open_fcv: [false, false],
+
+            apu_bleed_on: false,
+            crossbleed_is_on: false,
+            ditching_is_on: false,
+            engine_1_in_start_mode: false,
+            engine_2_in_start_mode: false,
+            engine_1_on_fire: false,
+            engine_2_on_fire: false,
+            flow_control_valve_open: false,
+            flow_selector_position: OvhdFlowSelector::Norm,
+            pack_in_start_condition: false,
+            single_pack_operation: false,
+        }
+    }
+
+    fn update(
+        &mut self,
+        aircraft_state: &AcStateManager,
+        acs_overhead: &AirConditioningSystemOverhead,
+        engines: [&impl EngineCorrectedN1; 2],
+        pressurization: &impl CabinAltitude,
+        pack_flow_valve: &[PackFlowValve; 2],
+    ) {
+        self.flow_control_valve_open = pack_flow_valve.iter().any(|fcv| fcv.fcv_is_open());
+        self.single_pack_operation =
+            pack_flow_valve[0].fcv_is_open() != pack_flow_valve[1].fcv_is_open();
+        self.pack_in_start_condition = self.pack_start_condition_determination(pack_flow_valve);
+        // TODO: Add overheat protection
+        self.flow_demand = self.flow_demand_determination(aircraft_state);
```

Comment:
**suggestion**: `pack_in_start_condition` can be passed as an argument, instead of being a field on the struct.
```suggestion
        let pack_in_start_condition = self.pack_start_condition_determination(pack_flow_valve);
        // TODO: Add overheat protection
        self.flow_demand = self.flow_demand_determination(aircraft_state, pack_in_start_condition);
```

Reply:
Yes good spot, instead of keeping it in the `update` function I've moved it inside of `flow_demand_determination` altogether.